Userland/Applications/CrashReporter/main.cpp

```diff
@@ -352,6 +270,60 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         }
     };
 
+    (void)Threading::BackgroundAction<ThreadBacktracesAndCpuRegisters>::construct(
+        [&, coredump = move(coredump)](auto&) {
+            ThreadBacktracesAndCpuRegisters results;
+            size_t thread_index = 0;
+            coredump->for_each_thread_info([&](auto& thread_info) {
+                results.thread_backtraces.append(build_backtrace(*coredump, thread_info, thread_index, [&](size_t frame_index, size_t frame_count) {
+                    window->set_progress(100.0f * (float)(frame_index + 1) / (float)frame_count);
+                    progressbar.set_value(frame_index + 1);
+                    progressbar.set_max(frame_count);
+                    Core::EventLoop::wake();
+                }));
+                results.thread_cpu_registers.append(build_cpu_registers(thread_info, thread_index));
+                ++thread_index;
+                return IterationDecision::Continue;
+            });
+            return results;
+        },
+        [&](auto results) {
+            for (auto& backtrace : results.thread_backtraces) {
+                auto container = MUST(backtrace_tab_widget->try_add_tab<GUI::Widget>(backtrace.title));
+                (void)MUST(container->template try_set_layout<GUI::VerticalBoxLayout>());
+                container->layout()->set_margins(4);
+                auto backtrace_text_editor = MUST(container->template try_add<GUI::TextEditor>());
+                backtrace_text_editor->set_text(backtrace.text);
+                backtrace_text_editor->set_mode(GUI::TextEditor::Mode::ReadOnly);
+                backtrace_text_editor->set_should_hide_unnecessary_scrollbars(true);
+            }
+
+            for (auto& cpu_registers : results.thread_cpu_registers) {
+                auto container = MUST(cpu_registers_tab_widget->try_add_tab<GUI::Widget>(cpu_registers.title));
+                (void)MUST(container->template try_set_layout<GUI::VerticalBoxLayout>());
+                container->layout()->set_margins(4);
+                auto cpu_registers_text_editor = MUST(container->template try_add<GUI::TextEditor>());
+                cpu_registers_text_editor->set_text(cpu_registers.text);
+                cpu_registers_text_editor->set_mode(GUI::TextEditor::Mode::ReadOnly);
+                cpu_registers_text_editor->set_should_hide_unnecessary_scrollbars(true);
+            }
+
+            progressbar.set_visible(false);
+            tab_widget.set_visible(true);
+            window->resize(460, 340);
+            window->set_progress(0);
+
+            MUST(Core::System::unveil(executable_path.characters(), "r"));
```

Comment:
All these MUSTs are quite sad. Does BackgroundAction support returning a type from the second lambda here? If we could get the ErrorOr all the way out to the executor of the background action... Well I'm not sure what we'd do with it other than return from main... Now I see what @sin-ack was going on about returning an ErrorOr from a callback for. 🤔

Reply:
Unfortunately the on_complete lambda must have return type void, hence the MUSTs. I considered modifying BackgroundAction to return a value from on_complete, but I'm not sure how to best accomplish that given it'll be executed on a different thread. Feels like a bigger change worthy of its own PR.

Reply:
`unveil()` calls are meant to be made as early as possible in the program's main function.
Deferring them until a thread completion callback is very strange. Please move them back to earlier in the program.

Reply:
Unveiling was deferred here because we need read access to shared libraries for symbolication. I was under the impression that it wasn't possible to know which shared libraries we'd need ahead of time, but after some digging I realized  `coredump->for_each_library` exists so it is possible and is done now in the latest push.